package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx

```diff
@@ -0,0 +1,317 @@
+      () => useLatestMessagePreview(CHANNEL_WITH_DELETED_MESSAGES, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        created_at: 'LT',
+        messageObject: {
+          type: 'deleted',
+        },
+        previews: [{ bold: false, text: 'Message deleted' }],
+        status: 0,
+      });
+    });
+  });
+
+  it('should return message deleted', async () => {
+    const { result } = renderHook(
+      () => useLatestMessagePreview(CHANNEL_WITH_NO_MESSAGES, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        created_at: 'LT',
+        previews: [{ bold: false, text: 'Nothing yet...' }],
+        status: 0,
+      });
+    });
+  });
+
+  it('should return latest text', async () => {
+    const { result } = renderHook(
+      () => useLatestMessagePreview(CHANNEL_WITH_MESSAGES_TEXT, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        created_at: 'L',
+        messageObject: {
+          args: 'string',
+          attachments: [],
+          channel: { data: { name: 'okechukwu' }, state: { messages: [] } },
+          cid: 'stridodong',
+          command: 'giphy',
+          command_info: {
+            name: 'string',
+          },
+          created_at: new Date('2021-02-12T12:12:35.862Z'),
+          deleted_at: new Date('2021-02-12T12:12:35.862Z'),
+          id: 'jbkjb',
+          text: 'jkbkbiubicbi',
+          type: 'MessageLabel',
+          user: {
+            id: 'okechukwu',
+          },
+        },
+        previews: [
+          { bold: true, text: '@okechukwu: ' },
+          { bold: false, text: 'jkbkbiubicbi' },
+        ],
+        status: 0,
+      });
+    });
+  });
+
+  it('should return a channel with an empty message', async () => {
+    const { result } = renderHook(
+      () => useLatestMessagePreview(CHANNEL_WITH_EMPTY_MESSAGE, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        created_at: 'L',
+        messageObject: {
+          args: 'string',
+          attachments: [],
+          cid: 'stridodong',
+          command_info: {
+            name: 'string',
+          },
+          created_at: new Date('2021-02-12T12:12:35.862Z'),
+          deleted_at: new Date('2021-02-12T12:12:35.862Z'),
+          mentioned_users: [
+            { id: 'Max', name: 'Max' },
+            { id: 'Ada', name: 'Ada' },
+            { id: 'Enzo', name: 'Enzo' },
+          ],
+        },
+        previews: [
+          { bold: false, text: '' },
+          { bold: false, text: 'Empty message...' },
+        ],
+        status: 0,
+      });
+    });
+  });
+
+  it('should return a channel with a mentioned user', async () => {
```

Comment:
same for the name here, do we return a channel here or actually a message preview with a mentioned user?

Reply:
Please be more spec. with assertion